LanguageFeatures/Patterns/record_A04_t01.dart

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion
+/// recordPattern         ::= '(' patternFields? ')'
+/// patternFields         ::= patternField ( ',' patternField )* ','?
+/// patternField          ::= ( identifier? ':' )? pattern
+/// ...
+/// It is a compile-time error if a name cannot be inferred for a named field
+/// pattern with the field name omitted
+///
+/// @description Checks that it is a compile-time error if a name cannot be
+/// inferred for a named field pattern with the field name omitted
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=patterns,records
+
+main() {
+  var (:n1) = (x1: 1);
```

Comment:
I think this test is impossible to write, unless the pattern can be refutable.

(Check out #2701 for a discussion about allowing refutable patterns in a declaration context).